File: lab5/DictionaryClient.c
/*
 *   DictionaryClient.c
 *  Test client for the Dictionary ADT
 */

#include<stdio.h>
#include<stdlib.h>
#include"Dictionary.h"

int main(int argc, char* argv[]){
   int v;
   DictionaryRef A = newDictionary();

    insert(A, 1, 111);
    insert(A, 2, 222);
    insert(A, 3, 333);
    insert(A, 4, 444);
    insert(A, 5, 555);
    insert(A, 6, 666);
    insert(A, 7, 777);
    printDictionary(A, stdout);
    v = lookup(A, 1);
    printf("key=1 "); v==UNDEF?printf("not found\n"):printf("value=%d\n", v);
    v = lookup(A, 3);
    printf("key=3 "); v==UNDEF?printf("not found\n"):printf("value=%d\n", v);
    v = lookup(A, 7);
    printf("key=7 "); v==UNDEF?printf("not found\n"):printf("value=%d\n", v);
    v = lookup(A, 8);
    printf("key=8 "); v==UNDEF?printf("not found\n"):printf("value=%d\n", v);
    printf("\n"); 

    /* insert(A, 2, 666); */ /* causes insert() duplicate keys error */
    
    delete(A, 1);
    delete(A, 3);
    delete(A, 7);
    printDictionary(A, stdout);
    
    /* delete(A, 8); */ /* causes delete() non-existent key error */
    
    printf("%s\n", isEmpty(A)?"true":"false");
    printf("%d\n", size(A));
    makeEmpty(A);
    printf("%s\n", isEmpty(A)?"true":"false");
    printDictionary(A, stdout);
    
    freeDictionary(&A);
    return(EXIT_SUCCESS);
}

File: lab7/DictionaryClient.java
// DictionaryClient.java
// Client module for testing Dictionary ADT

public class DictionaryClient{

   public static void main(String[] args){
      int v;
      Dictionary D = new Dictionary();

      D.insert(4, 35);
      D.insert(2, 32);
      D.insert(6, 17);
      D.insert(5, 44);
      D.insert(1, 41);
      D.insert(7, 79);
      D.insert(3, 64);
      
      System.out.println(D);

      v = D.lookup(1);
      System.out.println("key=1 " + (v==Dictionary.UNDEF?"not found":("value="+v)) );
      v = D.lookup(3);
      System.out.println("key=3 " + (v==Dictionary.UNDEF?"not found":("value="+v)) );
      v = D.lookup(7);
      System.out.println("key=7 " + (v==Dictionary.UNDEF?"not found":("value="+v)) );
      v = D.lookup(8);
      System.out.println("key=8 " + (v==Dictionary.UNDEF?"not found":("value="+v)) );
      System.out.println();

      // D.insert(2,32);  // causes KeyCollisionException

      D.delete(1);
      D.delete(3);
      D.delete(7);
      System.out.println(D);

      // D.delete(8);  // causes KeyNotFoundException

      System.out.println(D.isEmpty());
      System.out.println(D.size());
      D.makeEmpty();
      System.out.println(D.isEmpty());
      System.out.println(D);

   }
}